clients/web/src/views/body/CollectionView.js

```diff
@@ -156,6 +138,37 @@ var CollectionView = View.extend({
                 this.hierarchyWidget.refreshFolderList();
             }
         }, this);
+    },
+
+    deleteConfirmation: function () {
+        let params = {
+            text: 'Are you sure you want to delete the collection <b>' +
+                  this.model.escape('name') + this.model.escape('nFolders') + '</b>?',
+            yesText: 'Delete',
+            escapedHtml: true,
+            confirmCallback: _.bind(function () {
+                this.model.on('g:deleted', function () {
+                    events.trigger('g:alert', {
+                        icon: 'ok',
+                        text: 'Collection deleted.',
+                        type: 'success',
+                        timeout: 4000
+                    });
+                    router.navigate('collections', {trigger: true});
+                }).destroy();
+            }, this)
+        };
+        if (this.model.get('nFolders') !== 0 || this.model.get('size') !== 0) {
+            params = _.extend({
+                additionalText: '<b>' + this.model.escape('name') + '</b>' +
+                                ' contains <b>' + this.model.escape('nFolders') +
+                                ' folders</b> taking up <b>' +
+                                formatSize(this.model.escape('size')) + '</b>',
```

Comment:
`size` should be an integer, so this should be `formatSize(parseInt(this.model.get('size'), 10))`, using `parseInt` to guard against bad values rather than `escape`.